gcloud-java-storage/src/main/java/com/google/gcloud/storage/Storage.java

```diff
@@ -508,6 +526,26 @@ public static BlobSourceOption metagenerationMatch(long metageneration) {
     public static BlobSourceOption metagenerationNotMatch(long metageneration) {
       return new BlobSourceOption(StorageRpc.Option.IF_METAGENERATION_NOT_MATCH, metageneration);
     }
+
+    static BlobSourceOption[] setGeneration(BlobId blobId, Iterable<BlobSourceOption> options) {
```

Comment:
I am not sure why we need this.

Doesn't `StorageImpl.addToOptionMap` already takes care of it.
Wouldn't it use either the provided generation in the option or take the value from the BlobInfo or complain if option is specified but does not have a value?